tenant_schemas/storage.py

```diff
@@ -0,0 +1,45 @@
+import os
+
+from django.core.exceptions import SuspiciousOperation
+from django.utils._os import safe_join
+
+from django.db import connection
+
+from django.core.files.storage import FileSystemStorage
+
+__all__ = ('TenantStorageMixin', 'TenantFileSystemStorage')
+
+
+class TenantStorageMixin(object):
+    """
+    Mixin that can be combined with other Storage backends to colocate media
+    for all tenants in distinct subdirectories.
+
+    Using rewriting rules at the reverse proxy we can determine which content
```

Comment:
This needs to be documented with some practical examples. I've already done this on my own site so can extract that for public consumption.